kbfsedits/tlf_history.go

```diff
@@ -311,13 +311,13 @@ func (th *TlfHistory) recomputeLocked() (
 		}
 	}
 	sort.Sort(history)
-	// Garbage-collect any writers that don't appear in the history.
-	for i := maxWritersPerHistory; i < len(history); i++ {
-		delete(th.byWriter, history[i].writerName)
-		delete(writersWhoNeedMore, history[i].writerName)
-	}
 	if len(history) > maxWritersPerHistory {
 		history = history[:maxWritersPerHistory]
+		// Garbage-collect any writers that don't appear in the history.
+		for i := maxWritersPerHistory; i < len(history); i++ {
+			delete(th.byWriter, history[i].writerName)
+			delete(writersWhoNeedMore, history[i].writerName)
+		}
```

Comment:
I think this still needs to be above the `history = history[:maxWritersPerHistory]` right?

Reply:
Oops, excellent call.